File: front_app/ztmtad_do_tamtad/lib/path.dart
class Path {
  Path({
    required this.segments,
    required this.totalMinutesLenght,
    required this.arrivalTime,
  });

  final List<PathSegment> segments;
  final int totalMinutesLenght;
  final DateTime arrivalTime;
}

class PathSegment {
  PathSegment({
    required this.route,
    required this.departureTime,
    required this.minutesLength,
    required this.routeType,
    required this.stopDesc,
  });

  final String route;
  final DateTime departureTime;
  final int minutesLength;
  final RouteTypes routeType;
  final String stopDesc;
}

enum RouteTypes {
  bus,
  tram,
  ferry,
  train,
}

RouteTypes routeTypeFromString(String str) {
  switch (str) {
    case "BUS":
      return RouteTypes.bus;
    case "TRAM":
      return RouteTypes.tram;
    case "FERRY":
      return RouteTypes.ferry;
    case "TRAIN":
      return RouteTypes.train;
    default:
      return RouteTypes.tram;
  }
}

File: path.py
class Path:
    def __init__(self, total_time: int, path_segments, arrival_time: str):
        self.total_time = total_time
        self.path_segments = path_segments
        self.arrival_time = arrival_time # clocktime of arrival at the target

    def to_json(self):
        segment_jsons = []
        for p in self.path_segments:
            segment_jsons.append(p.to_json())
        return {'total_time': self.total_time, 'path_segments': segment_jsons, 'arrival_time': self.arrival_time}


class PathSegment:
    def __init__(self, route, route_type, minutes, departure_time, stop_desc):
        self.route = route,
        self.routeType = route_type,  # bus or train or what?
        self.minutes = minutes,  # how long does this segment take in minutes
        self.departureTime = departure_time,  # departure time
        self.stop_desc = stop_desc,  # stopDesc of departure stop for this path_segment

    def to_json(self):
        return {'route': self.route,
                'route_type': self.routeType,
                'minutes': self.minutes,
                'departure_time': self.departureTime,
                'stop_desc': self.stop_desc}
